adapters/handlers/graphql/local/get/class_builder_fields.go

```diff
@@ -372,6 +372,9 @@ func (r *resolver) makeResolveGetClass(className string) graphql.FieldResolveFn
 		// extracts bm25 (sparseSearch) from the query
 		var keywordRankingParams *searchparams.KeywordRanking
 		if bm25, ok := p.Args["bm25"]; ok {
+			if len(sort) > 0 {
+				return nil, fmt.Errorf("bm25 search is not compatible with sort")
+			}
```

Comment:
I think that all of our validation logic for GraphQL we place here in [explorer.GetClass](https://github.com/weaviate/weaviate/blob/master/usecases/traverser/explorer.go#L108-L139) method

Reply:
The benefit of having the checks here, is that the GQL query will fail to resolve with this incompatible set of parameters. If we move the checks to `(Explorer) GetClass`, then the query will resolve successfully, when it could have been determined ahead of resolution that query is invalid.

That being said, I don't think there is a huge practical difference during usage, so if you still want this change, I can do so